index.js

```diff
@@ -168,7 +168,16 @@ HttpsProxyAgent.prototype.callback = function connect(req, opts, fn) {
 			//
 			// See: https://hackerone.com/reports/541502
 			socket.destroy();
-			socket = new events.EventEmitter();
+			socket = new stream.Duplex({
+				read() {},
+				write(chunk, encoding, callback) {
+					callback();
+				}
+			});
+
+			if (process.versions.modules === '48') {
```

Comment:
I'm not very happy with this. I wonder if it's better to use an actual `Socket`.

Reply:
I've used a new and closed `net.Socket` in f6e253806f0898a8d213b2b3650153ca1527f1f1. Let me know if I should squash it, drop it, or keep it as it.

Reply:
I'll squash when merging the PR, don't worry about that part for now.